blog/content/post/2022-12-22-fobnail_application.md

```diff
@@ -0,0 +1,357 @@
+codes produced by Fobnail Token.
+
+## Building and running
+
+Building hasn't change much since [last time](https://blog.3mdeb.com/2022/2022-05-25-fobnail_provisioning/#building-and-running).
+There are some changes done to produce `fobnail-attester-with-provisioning`
+along with non-provisioning version, but instructions for building didn't
+change. The process is now [part of Fobnail documentation](https://fobnail.3mdeb.com/building/).
+
+Ubuntu LTS 22.04 and tools from official packages included in that distribution
+were used for writing and testing of this blog post. This includes building,
+flashing and use of all tools, as they were not performed on the same machine.
+The following revisions of components were used:
+
+- [Fobnail SDK](https://github.com/fobnail/fobnail-sdk/commit/f477d804859d58c63ccd83be487423739591adb6)
+- [Fobnail](https://github.com/fobnail/fobnail/commit/4f11662002f611187085b072a4b31288991a6f73)
+- [Fobnail Attester](https://github.com/fobnail/fobnail-attester/commit/781389945e34b6bdbcf2d615269552fb6413dceb)
+- [Fobnail Platform Owner](https://github.com/fobnail/fobnail-platform-owner/commit/2af43f26870d1968429678e2adcbf90142b0b330)
+
+Despite changes in the architecture, applications are started mostly the same as
+before. Main difference is that the applications don't have to (and shouldn't)
+be already running when Fobnail Token is being plugged in. Attester changed a
+bit more, it now can take arguments which use Fobnail Token Services.
+Description of those can be obtained by starting Attester with `--help`:
+
+```
+Usage:
+    fobnail-attester [CMD]...
+
+Commands:
+    --read-file token_fname:local_fname
+    -r token_fname:local_fname
+        Read file token_fname from Fobnail Token and save it in local_fname.
+
+    --write-file local_fname:token_fname
+    -w local_fname:token_fname
+        Write file local_fname to token_fname on Fobnail Token.
+
+    --delete-file token_fname
+    -d token_fname
+        Remove file token_fname from Fobnail Token.
+
+All commands are executed only if the attestation was successful.
+
+Both --read-file and --write-file can take '-' as local_fname to use stdout
+and stdin, respectively. If '-' appears in multiple write commands, the same
+data is written to all files. If '-' appears in multiple read commands, output
+consists of concatenated content of all files, in the order in which they are
+read. Use './-' to access regular file named '-'.
+
+Multiple commands may be specified at once, in that case they are executed in
+order in which they appear on the command line. Attestation is performed only
+once. If any of the commands fails, further commands are not executed.
+```
+
+## Preparing encrypted disk image for use with Fobnail
+
+Some preparations have to be done during provisioning by administrator to make
+user's life as easy as possible. Following steps were performed on Ubuntu
+**TBD**, they may be slightly different for other versions and distributions.
```

Comment:
@krystian-hebel TBD?

Reply:
For confirmation: Ubuntu 22.04

Reply:
Ack, will add this together with link to demo when it's ready.

Reply:
Added.